src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -188,14 +189,46 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
         }
 
         [Fact]
-        public void Create_Directory_From_DirectoryInfo()
+        public void DirectoryInfo_Create_NullDirectorySecurity()
+        {
+            DirectoryInfo info = new DirectoryInfo("path");
+            Assert.Throws<ArgumentNullException>(() => info.Create(null));
+        }
+
+
+        [Fact]
+        public void DirectoryInfo_Create_DefaultDirectorySecurity()
+        {
+            DirectorySecurity security = new DirectorySecurity();
+            CreateAndVerifyDirectoryWithSecurity(security);
+        }
+        [Theory]
+        [InlineData("S-1-5-32-544", FileSystemRights.FullControl, AccessControlType.Allow)]
+        public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(string sddlForm, FileSystemRights rights, AccessControlType controlType)
+        {
+            DirectorySecurity security = new DirectorySecurity();
+            SecurityIdentifier identity = new SecurityIdentifier(sddlForm);
+            FileSystemAccessRule accessRule = new FileSystemAccessRule(identity, rights, controlType);
+
+            security.AddAccessRule(accessRule);
+
+            CreateAndVerifyDirectoryWithSecurity(security);
+        }
+
+        private void CreateAndVerifyDirectoryWithSecurity(DirectorySecurity security)
         {
             using var directory = new TempDirectory();
+
             string path = Path.Combine(directory.Path, "directory");
             DirectoryInfo info = new DirectoryInfo(path);
-            DirectorySecurity security = new DirectorySecurity();
+
             info.Create(security);
+
             Assert.True(Directory.Exists(path));
+            Assert.Equal(typeof(FileSystemRights), security.AccessRightType);
+
+            DirectorySecurity retrievedSecurity = info.GetAccessControl();
+            Assert.Equal(typeof(FileSystemRights), retrievedSecurity.AccessRightType);
```

Comment:
Why aren't you actually verifying the rights on the object?

Reply:
I was investigating what exactly to compare. I pushed another commit. Can you please review the changes?

Reply:
I just saw your other comment about using WellKnownSidType. I'll address that.

Reply:
Fixed.